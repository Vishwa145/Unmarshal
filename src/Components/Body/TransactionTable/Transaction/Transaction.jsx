import React from "react";

function Transaction({ transaction }) {
  //Later to be updated with result of the calculations or values from trasactions.
  const change = 1.92 + "%";
  const currentvalue = "$" + (7641.75).toLocaleString("en-US");
  const balance = (43761.121).toLocaleString("en-US") + " POND . $" + 0.17;
  const profitpercent = 2.32;
  const profit =
    (profitpercent >= 0 ? "+" : "-") +
    profitpercent +
    "%" +
    "($" +
    173.13 +
    ")";

  return (
    <div className="Content">
      <img src={transaction.logo_url} alt="" />
      <div className="TransactionDetails">
        <div>
          <p>
            {transaction.contract_name}{" "}
            <span className="Highlight">{change}</span>
          </p>
          <p>{currentvalue}</p>
        </div>
        <div>
          <p style={{ color: "#576574" }}>{balance}</p>
          <p style={{ color: "#1dd1a1" }}>{profit}</p>
        </div>
      </div>
    </div>
  );
}

export default Transaction;
