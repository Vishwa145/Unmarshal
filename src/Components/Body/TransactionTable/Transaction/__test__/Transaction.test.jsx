import React from "react";
import ReactDOM from "react-dom";
import Transaction from "../Transaction";
import {cleanup} from "@testing-library/react";
import renderer from "react-test-renderer"

const transaction1 = {
    contract_name: "USD Coin",
    contract_ticker_symbol: "USDC",
    contract_decimals: 6,
    contract_address: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
    coin: 60,
    type: "ERC20",
    balance: "20247689228",
    quote: 20247.689228,
    quote_rate: 1,
    logo_url: "https://assets.unmarshal.io/tokens/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48.png"
};

const transaction2 = {
    contract_name: "ETH Price Action Candlestick Set",
    contract_ticker_symbol: "ETHPA",
    contract_decimals: 18,
    contract_address: "0x09E4BDFb273245063eF5E800D891eFF7d04f9B83",
    coin: 60,
    type: "ERC20",
    balance: "100000000000000000",
    quote: 234.51500000000001,
    quote_rate: 2345.15,
    logo_url: "https://assets.unmarshal.io/tokens/0x09E4BDFb273245063eF5E800D891eFF7d04f9B83.png"
};

afterEach(cleanup);
it("render successfully", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Transaction transaction={transaction1}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("render successfully", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Transaction transaction={transaction2}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Transaction transaction={transaction1}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<Transaction transaction={transaction2}/>).toJSON();
    expect(tree).toMatchSnapshot();
});