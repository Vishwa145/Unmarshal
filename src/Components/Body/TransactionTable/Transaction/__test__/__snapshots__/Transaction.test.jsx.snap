// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="Content"
>
  <img
    alt=""
    src="https://assets.unmarshal.io/tokens/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48.png"
  />
  <div
    className="TransactionDetails"
  >
    <div>
      <p>
        USD Coin
         
        <span
          className="Highlight"
        >
          1.92%
        </span>
      </p>
      <p>
        $7,641.75
      </p>
    </div>
    <div>
      <p
        style={
          Object {
            "color": "#576574",
          }
        }
      >
        43,761.121 POND . $0.17
      </p>
      <p
        style={
          Object {
            "color": "#1dd1a1",
          }
        }
      >
        +2.32%($173.13)
      </p>
    </div>
  </div>
</div>
`;

exports[`Matches snapshot 2`] = `
<div
  className="Content"
>
  <img
    alt=""
    src="https://assets.unmarshal.io/tokens/0x09E4BDFb273245063eF5E800D891eFF7d04f9B83.png"
  />
  <div
    className="TransactionDetails"
  >
    <div>
      <p>
        ETH Price Action Candlestick Set
         
        <span
          className="Highlight"
        >
          1.92%
        </span>
      </p>
      <p>
        $7,641.75
      </p>
    </div>
    <div>
      <p
        style={
          Object {
            "color": "#576574",
          }
        }
      >
        43,761.121 POND . $0.17
      </p>
      <p
        style={
          Object {
            "color": "#1dd1a1",
          }
        }
      >
        +2.32%($173.13)
      </p>
    </div>
  </div>
</div>
`;
